### PR TITLE
chore(test): validate the commit messages on travis

### LINKED ISSUE #158: contribution: check PR is done on develop
We should ensure that our PRS follows this:
- [ ] pull request is done on develop

Travis is providing the targeted branch in env variable "TRAVIS_BRANCH" when in a PR.

### PR DESCRIPTION
**Pull request by redox**
Ref #158

### CODE REVIEW

**Comment by vvo** [comment]
Maybe we should validate the commit message format only in after_success from travis. Thus allowing pushes to check for passing tests before checking for the commit message format. Should be better?

**Comment by redox** [comment]
> Maybe we should validate the commit message format only in after_success from travis. Thus allowing pushes to check for passing tests before checking for the commit message format. Should be better?

The issue with `after_success` is that it doesn't affect the build result. I think it's OK to fail because of a commit message and without running the tests, no?

**Comment by vvo** [comment]
> I think it's OK to fail because of a commit message and without running the tests, no?

It will be a pain for developers in the end, maybe we can just run the check AFTER the tests, that was initially what I wanted to do.

Knowing that the tests are passing here is way more important than having the commit message check failing right away (a bit like asking for lint before tests, it's a pain).

I wish travis exposed ways to have multiple status checks (https://github.com/blog/1227-commit-status-api) but no. The only way to have something better is to make a contribution bot run and do checks inside various status checks. This way you can see that the tests are passing but not the commit check.

For now, checking the commit status after the tests should be enough to keep a good flow.

**Comment by redox** [comment]
> For now, checking the commit status after the tests should be enough to keep a good flow.

Works for me, updated!

File: scripts/validate-commit-msgs.sh [hunk 1]
```diff
@@ -0,0 +1,46 @@
+#! /usr/bin/env bash
+
+# Checks the commits msgs in the range of commits travis is testing.
+# Based heavily on
+# https://raw.githubusercontent.com/angular/angular.js/291d7c467fba51a9cb89cbeee62202d51fe64b09/validate-commit-msg.js
```
Comment (vvo) [inline scripts/validate-commit-msgs.sh]: Nice :)

**Comment by vvo** [comment]
also that's awesome you did it :)

**Comment by pixelastic** [comment]
Just a note: you can also use local git hooks to test for commit messages and tests before pushing.

**Comment by redox** [comment]
> Just a note: you can also use local git hooks to test for commit messages and tests before pushing.

Yes, that's what Angular is doing with a "initialization" script to run setting it up after cloning the repo.

**Comment by vvo** [comment]
> Yes, that's what Angular is doing with a "initialization" script to run setting it up after cloning the repo.

Yes but I think I saw somewhere they wanted to do it in travis also

**Pull request merged**